reVX/setbacks/base.py

```diff
@@ -0,0 +1,658 @@
+# -*- coding: utf-8 -*-
+"""
+Compute setbacks exclusions
+"""
+from abc import ABC
+from concurrent.futures import as_completed
+from warnings import warn
+import os
+import logging
+import pathlib
+import numpy as np
+import geopandas as gpd
+from rasterio import features
+from shapely.geometry import shape
+
+from rex.utilities import parse_table, SpawnProcessPool, log_mem
+from reV.handlers.exclusions import ExclusionLayers
+from reVX.utilities.exclusions_converter import ExclusionsConverter
+from reVX.utilities.utilities import log_versions
+
+logger = logging.getLogger(__name__)
+
+
+class BaseSetbacks(ABC):
+    """
+    Create exclusions layers for setbacks
+    """
+
+    def __init__(self, excl_fpath, plant_height, regulations_fpath=None,
+                 multiplier=None, hsds=False, chunks=(128, 128)):
+        """
+        Parameters
+        ----------
+        excl_fpath : str
+            Path to .h5 file containing exclusion layers, will also be the
+            location of any new setback layers
+        plant_height : float | int
+            Plant height (m), used to determine setback distance.
```

Comment:
I guess I just mimicked Anthony's nomenclature: "C) include ability to model static or height based setbacks (e.g., 30-meters or 1.1x height of plant).". I even asked him recently what plant height I should use for a test run and he didn't mention anything about me using the wrong term (maybe he was just being polite lolol). 

I think plant height also matches the user config nicely. The wind one looks something like
```
...
"hub_height": 135,
"rotor_diameter": 200
...
```

while the solar one would look like 

```
...
"plant_height": 5
...
```

But in any case, I'm not tied to one way or another. I will say that changing the name to `base_setback` would definitely require some effort, as the cli and config file will also have to be updated. That being said, its totally doable, so let me know if you think that's worth it and I will work on that.

Reply:
I think it might be worth it... In this case the height of the solar panel is not 5 meters, you're just modeling a 5 meter setback for the plant. Same with the 30m/1.1x example. The height of the plant is not 30 meters, its just a generic 30 meter setback.

Reply:
I may be wrong about this, let's confirm with anthony

Reply:
So where do we stand on this?

Reply:
okay can you set this variable to `base_setback_dist`? and also change the entry in the solar config? Painful but necessary IMO

Reply:
On it!